commit 2634448683d32d131df5d050f07c6dec4aeb5d39
Author: Jason Kridner <jkridner@gmail.com>
Date:   Fri Jan 11 10:02:13 2019 -0500

    pinctrl: mcp23s08: spi: Fix regmap allocation for mcp23s18
    
    [ Upstream commit f165988b77ef849eb0c1aebd94fe778024f88314 ]
    
    Fixes issue created by 9b3e4207661e67f04c72af15e29f74cd944f5964.
    
    It wasn't possible for one_regmap_config to be non-NULL at the point
    it was tested for mcp23s18 devices.
    
    Applied the same pattern of allocating one_regmap_config using
    devm_kmemdump() and then initializing the local regmap structure
    from that.
    
    Signed-off-by: Jason Kridner <jdk@ti.com>
    Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/pinctrl/pinctrl-mcp23s08.c b/drivers/pinctrl/pinctrl-mcp23s08.c
index cf73a403d22d..cecbce21d01f 100644
--- a/drivers/pinctrl/pinctrl-mcp23s08.c
+++ b/drivers/pinctrl/pinctrl-mcp23s08.c
@@ -832,8 +832,13 @@ static int mcp23s08_probe_one(struct mcp23s08 *mcp, struct device *dev,
 		break;
 
 	case MCP_TYPE_S18:
+		one_regmap_config =
+			devm_kmemdup(dev, &mcp23x17_regmap,
+				sizeof(struct regmap_config), GFP_KERNEL);
+		if (!one_regmap_config)
+			return -ENOMEM;
 		mcp->regmap = devm_regmap_init(dev, &mcp23sxx_spi_regmap, mcp,
-					       &mcp23x17_regmap);
+					       one_regmap_config);
 		mcp->reg_shift = 1;
 		mcp->chip.ngpio = 16;
 		mcp->chip.label = "mcp23s18";
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1071_2634448683d32d131df5d050f07c6dec4aeb5d39.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1071_2634448683d32d131df5d050f07c6dec4aeb5d39.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
